openmdao/core/group.py

```diff
@@ -1364,8 +1364,8 @@ def _setup_connections(self, recurse=True):
                                         msg = f"{self.msginfo}: The source indices " + \
                                               f"do not specify a valid index for the " + \
                                               f"connection '{abs_out}' to '{abs_in}'. " + \
-                                              f"Index '{i}' is out of range for source " + \
-                                              f"dimension of size {d_size}."
+                                              f"src_indices index: {i} must out of range for " + \
```

Comment:
typo:  'must'.  Did you intend 'must be'?  I think the new form of the error message is confusing.  It's confusing at the end where it says that the index {i} is out of range for the input vector.  `src_indices` reference into the source vector, not the input vector, and `d_size` is the size of the current source dimension.  I don't think the original issue mentioned any problems with the form of the error message, did it?

Reply:
Yeah that was a typo. 
I'm open to leaving the error message alone. To me the original error message didn't seem very clear. The original story was asking to raise an error if `src_indices` was out of bounds of the input vector, but this error already catches that.

Reply:
The `src_indices` reference into the source vector, not the input vector, so, for example, you could have an input vector of size 1 with a `src_indices` of [99], and as long as the source vector is of size 100 or greater, it's fine.  It is true though that the *number* of indices contained in `src_indices` must match the input size.  I added Ben as a requested reviewer for this one because I'm not sure we're addressing his actual issue.  I think he was talking about a situation where DefaultVectors are used under MPI.

Reply:
@DKilkenny : I agree with Bret that the old messages are more clear.

Per Bret's suggestion, you could try running the test you showed me under MPI (without adding any distributed comps or parallelgroups to the model) and see if a correct error is raised.

Reply:
I confirmed that the error is still raised under those conditions